modules/usage/module.php

```diff
@@ -539,7 +539,7 @@ private function get_elements_usage( $elements ) {
 
 				$changed_controls_count = $this->add_controls( $settings_controls, $element_controls, $element_ref );
 
-				$percent = $changed_controls_count / ( count( $element_controls ) / 100 );
+				$percent = ! empty( $element_controls ) ? $changed_controls_count / ( count( $element_controls ) / 100 ) : 0;
```

Comment:
I think it will work without it.
Maybe we can check that we don't mess with the usage data in general:
any usage data that is related to the controls will be miscalculated.

Reply:
I agree with the general claim, but the thing is that it causes some errors upon save, thus failing the save entirely -

```
Fatal error: Uncaught Error: Division by zero
in /Users/ronros/elementor/modules/usage/module.php on line 542

Call stack:

Elementor\M\U\Module::Elementor\M\U\{closure}()
call_user_func()
/Users/ronros/elementor/includes/db.php:264
Elementor\DB::iterate_data()
/Users/ronros/elementor/includes/db.php:268
Elementor\DB::iterate_data()
/Users/ronros/elementor/includes/db.php:261
Elementor\DB::iterate_data()
/Users/ronros/elementor/includes/db.php:268
Elementor\DB::iterate_data()
/Users/ronros/elementor/includes/db.php:261
Elementor\DB::iterate_data()
/Users/ronros/elementor/includes/db.php:268
Elementor\DB::iterate_data()
/Users/ronros/elementor/modules/usage/module.php:548
Elementor\M\U\Module::get_elements_usage()
/Users/ronros/elementor/modules/usage/module.php:570
Elementor\M\U\Module::save_document_usage()
/Users/ronros/elementor/modules/usage/module.php:183
Elementor\M\U\Module::after_document_save()
wp-includes/class-wp-hook.php:326
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:348
WP_Hook::do_action()
wp-includes/plugin.php:517
do_action()
/Users/ronros/elementor/core/base/document.php:881
Elementor\C\B\Document::save()
/Users/ronros/elementor/core/documents-manager.php:549
Elementor\Core\Documents_Manager::ajax_save()
call_user_func()
/Users/ronros/elementor/core/common/modules/ajax/module.php:176
Elementor\C\C\M\A\Module::handle_ajax_request()
wp-includes/class-wp-hook.php:324
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:348
WP_Hook::do_action()
wp-includes/plugin.php:517
do_action()
wp-admin/admin-ajax.php:192
```